### PR TITLE
Drop Node.js 12 Support

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

https://github.com/prettier/prettier/issues/10157#issuecomment-1185483173

## Checklist

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
There should .more work to do, see #11830

File: scripts/build/build.mjs [hunk 1]
```diff
@@ -133,7 +133,7 @@ async function createBundle(bundleConfig, options) {
 async function preparePackage() {
   const packageJson = await readJson(path.join(PROJECT_ROOT, "package.json"));
   packageJson.bin = "./bin-prettier.js";
-  packageJson.engines.node = ">=12.17.0";
+  packageJson.engines.node = ">=14.20.0";
```
Comment (fisker) [inline scripts/build/build.mjs:136]: We can comment out this line, use the value in package.json directory.

File: .github/workflows/prod-test.yml [hunk 2]
```diff
@@ -71,7 +71,6 @@ jobs:
           - "18"
           - "16"
           - "14"
-          - "12"
```
Comment (fisker) [inline .github/workflows/prod-test.yml:74]: We need run test on Node.js 14, please update `exclude` bellow.

**Review by fisker — APPROVED** [review]

File: changelog_unreleased/api/11830.md [hunk 3]
```diff
@@ -1,3 +1,3 @@
-#### [Breaking] Drop support for Node.js 10 (#11830 by @fisker)
+#### [Breaking] Drop support for Node.js 10 and 12 (#11830 by @fisker, #13118 by @sosukesuzuki)
 
-The minimal required Node.js version is v12.17.0
+The minimal required Node.js version is v14.20.0
```
Comment (sosukesuzuki) [inline changelog_unreleased/api/11830.md:3]:
```suggestion
The minimal required Node.js version is v14
```

**Pull request merged**